### ISSUE TITLE
7.6.0 false positives with LESS detached rulesets

### ISSUE THREAD

**Issue by thybzi** [stylelint/stylelint#2088]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

**Version 7.6.0** introduces some bug when working with LESS ruleset variables: http://lesscss.org/features/#detached-rulesets-feature.

**LESS variable assignments for detached rulesets seem to be interpreted as CSS selectors**, and thus lead to some false positives (the problem doesn't exist in 7.5.0).

> Which rule, if any, is this issue related to?

* `selector-type-no-unknown`
* `rule-non-nested-empty-line-before`

> What CSS is needed to reproduce this issue?

```less
@welcome__logo-size_desktop: {
    width: 217px;
    height: 173px;
};
@welcome__side-triangle-size_desktop: {
    width: 243px;
    height: 286px;
};

```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "selector-type-no-unknown": [true, {
            "ignoreTypes": ["when", "and"]
        }],
        "rule-non-nested-empty-line-before": [ "always", {
            "ignore": ["after-comment"]
        }]
    }
}
```

> Which version of stylelint are you using?

`7.6.0`
(the problem doesn't exist in 7.5.0)

> How are you running stylelint: CLI, PostCSS plugin, Node API?

```
grunt stylelint
```
with grunt-stylelint 0.6.0 and the following section in Gruntfile.js
```js
        stylelint: {
            all: {
                src: 'resources/less/**/*.less'
            }
        },
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it is related with LESS detached rulesets feature syntax: http://lesscss.org/features/#detached-rulesets-feature

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
04:42:15 >> resources/less/player/splash/splash__globals.less
04:42:15 >>   3:1  ✖  Unexpected unknown type selector "@welcome__logo-size_desktop:"            selector-type-no-unknown         
04:42:15 >>   7:1  ✖  Unexpected unknown type selector "@welcome__side-triangle-size_desktop:"   selector-type-no-unknown         
04:42:15 >>   7:1  ✖  Expected empty line before non-nested rule                                 rule-non-nested-empty-line-before     
... (and so on with other variables in this file and other ones with ruleset assignments)
```

**Comment by evilebottnawi** [comment]
reproduce, wip

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2089
